Orange/data/variable.py

```diff
@@ -13,7 +13,7 @@
 # For storing unknowns
 Unknown = ValueUnknown = float("nan")
 # For checking for unknowns
-MISSING_VALUES = {"?", ".", "", "NA", "~", None}
+MISSING_VALUES = {"?", "nan", ".", "", "NA", "~", None}
```

Comment:
Without this, `StringVariable`s that had "nan" values considered them as values, not missing.